pmtiles/intro.qmd

```diff
@@ -0,0 +1,97 @@
+
+### Hilbert-oriented tiles
+
+Interally, tiles are oriented along a Hilbert Curve. This means that tiles that are spatially near each other are also located near each other in the file structure.
+
+This is especially appropriate for PMTiles because visualization purposes most often request data within a specific geographic area. Because spatially-nearby tiles are likely to be nearby in the file as well, this allows the PMTiles client to merge multiple requests for tiles into one larger request, rather than needing to fetch a different area of the file for each tile.
+
+## Multiple resolution
+
+PMTiles archives support storing a full XYZ pyramid of tile data. This means that you can store multiple zoom levels of data inside a single file.
+
+## Internal compression
+
+PMTiles allows tiles to be stored in the file with compression.
+
+## Generating PMTiles
+
+### From vector data
+
+The easiest way to generate PMTiles for vector data is through the [tippecanoe](https://github.com/felt/tippecanoe) tool. This will generate vector tiles that are ideal for visualization, removing small features at low zoom levels to keep tiles a manageable size.
+
+### From existing tiles or MBTiles
+
+PMTiles has a [command-line program](https://github.com/protomaps/go-pmtiles) for creating PMTiles if you already have an MBTiles file or a directory of tiles.
+
+## Using PMTiles
+
+### PMTiles viewer
+
+If you have an existing PMTiles archive, either as a local file or hosted on cloud storage, you can use the [PMTiles Viewer](https://protomaps.github.io/PMTiles/) to inspect the tiles hosted within the file.
+
+### JavaScript
+
+PMTiles doesn't have a standalone JavaScript library, but rather is designed to be used in conjunction with a JavaScript map rendering library.
+
+See the docs on viewing PMTiles in [Leaflet](https://protomaps.com/docs/frontends/leaflet), [MapLibre GL JS](https://protomaps.com/docs/frontends/maplibre) and [OpenLayers](https://protomaps.com/docs/frontends/openlayers).
+
+### Python
+
+PMTiles has a [Python package](https://github.com/protomaps/PMTiles/tree/main/python), which allows reading and writing PMTiles archives from Python.
+
+## Alternatives
+
+### PMTiles
+
+The most common alternative for PMTiles is [MBTiles](https://github.com/mapbox/mbtiles-spec), which was in many ways the precursor to PMTiles. MBTiles stores the included vector tiles in a table in a SQLite database. This has the benefit of being much easier to use than manually managing millions of tiny, individual files, but has the downside of not being serverless. In general, it's impossible to read from a SQLite database without fetching the entire file's content. This meant that frontend clients like a web browser couldn't fetch tiles directly using range requests, but rather a server has to be running to fetch tiles from the MBTiles file.
+
+### Saving individual files to cloud storage
```

Comment:
This is a little confusing because it's not really clear earlier in the document that pmtiles is an archive but in theory you can work with the "bare tiled data" itself. Could we add something earlier in the document about how the archive is comprised of many individual tiles and what format those files are in? I'm actually not sure myself.

Reply:
I tried to make this a bit clearer in https://github.com/developmentseed/cloud-optimized-geospatial-formats-guide/pull/40/commits/ebc58dcb813b312338f913809198f615bc410f4b by likening a PMTiles to a zip file